File: src/main/java/ru/ifmo/is/lab1/dragons/dto/DragonDto.java
package ru.ifmo.is.lab1.dragons.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;
import ru.ifmo.is.lab1.common.framework.dto.AuditableDto;
import ru.ifmo.is.lab1.coordinates.Coordinate;
import ru.ifmo.is.lab1.dragoncaves.DragonCave;
import ru.ifmo.is.lab1.dragonheads.DragonHead;
import ru.ifmo.is.lab1.dragons.DragonType;
import ru.ifmo.is.lab1.people.Person;

@Data
@EqualsAndHashCode(callSuper = true)
public class DragonDto extends AuditableDto {
  private int id;
  private String name;
  private Coordinate coordinates;
  private DragonCave cave;
  private Person killer;
  private DragonHead head;
  private DragonType type;
  private Integer age;
  private Integer wingspan;
  private Boolean speaking;
}

File: frontend/src/interfaces/dto/dragons/DragonDto.ts
import { AuditableDto } from '../AuditableDto';
import { CoordinateDto } from '../coordinates/CoordinateDto';
import { DragonCaveDto } from '../dragoncaves/DragonCaveDto';
import { DragonHeadDto } from '../dragonheads/DragonHeadDto';
import { DragonType } from '../../models/DragonType';
import { PersonDto } from '../people/PersonDto';

export interface DragonDto extends AuditableDto {
    name: string;
    coordinates: CoordinateDto;
    cave?: DragonCaveDto;
    killer?: PersonDto;
    head: DragonHeadDto;
    type: DragonType;
    age?: number;
    wingspan?: number;
    speaking?: boolean;
}
